lib/aws/lex/conversation/type/session_attributes.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Aws
+  module Lex
+    class Conversation
+      module Type
+        class SessionAttributes < Hash
+          include Base
+
+          optional :checkpoints
+
+          def checkpoints
+            @checkpoints ||= begin # rubocop:disable Style/RedundantBegin
+              JSON.parse(Base64.decode64(fetch(:checkpoints) { Base64.encode64([].to_json) })).map do |checkpoint|
```

Comment:
What do you think about using `Base64.urlsafe_encode64(..., padding: false)`?

We'd also likely need to use `Base64.urlsafe_decode64(...)`, but it doesn't need a `padding` flag at all.

I've noticed that Ruby's standard Base64 encoding tends to break up the output with newline characters. During our previous tests with AWS Lambda, I think we received string encoding errors during the serialization phase.